### DISCUSSION TITLE
AqualinkD service not starting

### DISCUSSION THREAD

**Discussion by mrdavidscott** [aqualinkd/AqualinkD#326] [Q&A] [answered]
Hello,

AqualinkD service won't start. The service fails to load. Here's the log:

Linux pi 6.6.20+rpt-rpi-v6 #1 Raspbian 1:6.6.20-1+rpt1 (2024-03-07) armv6l
× aqualinkd.service - Aqualink RS daemon
     Loaded: loaded (/etc/systemd/system/aqualinkd.service; enabled; preset: enabled)
     Active: failed (Result: protocol) since Wed 2024-08-07 01:37:49 PDT; 6h ago
    Process: 553 ExecStart=/usr/local/bin/aqualinkd $OPTS (code=exited, status=0/SUCCESS)
        CPU: 40ms

Aug 07 01:37:49 pi aqualinkd[553]: AqualinkD: Config BTN Aux_4         = label Spa Booster Pump |
Aug 07 01:37:49 pi aqualinkd[553]: AqualinkD: Config BTN Aux_5         = label Floor Cleaner   |
Aug 07 01:37:49 pi aqualinkd[553]: AqualinkD: Config BTN Aux_6         = label NONE            |
Aug 07 01:37:49 pi aqualinkd[553]: AqualinkD: Config BTN Aux_7         = label NONE            |
Aug 07 01:37:49 pi aqualinkd[553]: AqualinkD: Config BTN Pool_Heater   = label Pool Heater     |
Aug 07 01:37:49 pi aqualinkd[553]: AqualinkD: Config BTN Spa_Heater    = label Spa Heater      |
Aug 07 01:37:49 pi aqualinkd[553]: AqualinkD: Config BTN Solar_Heater  = label NONE            |
Aug 07 01:37:49 pi systemd[1]: aqualinkd.service: New main PID 555 does not exist or is a zombie.
Aug 07 01:37:49 pi systemd[1]: aqualinkd.service: Failed with result 'protocol'.
Aug 07 01:37:49 pi systemd[1]: Failed to start aqualinkd.service - Aqualink RS daemon.

This is what I enter:
sudo systemctl enable aqualinkd

I am however able to start AqualinkD with

sudo systemctl start aqualinkd

I wanted it to run as a service though so it will restart on boot.

**Comment by mrdavidscott** [comment, marked as answer]
Hello so I fixed it. Just thought I'd leave the solution here. So it was a problem with the service starting to Soon after a boot up. So I added these lines

[Unit]
Requires=network-online.target
After=network-online.target

[Service]
ExecStartPre=/bin/sleep 30


To the 
/etc/systemd/system/sshd.service 
File.

This also makes services wait for network to be online.

**Comment by sfeakes** [comment]
Thanks for posting the fix.  I'll look to use network-online.target vs the current network.target in future release.  I seem to remember network-online.target was very nit-n-miss on early raspbian distro's.

**Reply by mrdavidscott** [reply]
To be fair I'm not sure which line of code actually fixed it. I also added:
 
ExecStartPre=/bin/sleep 30

Which gives 30 seconds from boot before starting service.  But it works now and if there is a power failure everything will restart on its own, so I'm happy. Thank you for making this possible!
